src/main/java/com/google/jenkins/plugins/computeengine/NoDelayProvisionerStrategy.java

```diff
@@ -0,0 +1,109 @@
+      LOGGER.log(Level.FINE, "Provisioning not complete, NoDelayProvisionerStrategy is disabled");
+      return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+    }
+    final Label label = strategyState.getLabel();
+
+    LoadStatistics.LoadStatisticsSnapshot snapshot = strategyState.getSnapshot();
+    int availableCapacity =
+    snapshot.getAvailableExecutors()   // live executors
+            + snapshot.getConnectingExecutors()  // executors present but not yet connected
+            + strategyState.getPlannedCapacitySnapshot()     // capacity added by previous strategies from previous rounds
+            + strategyState.getAdditionalPlannedCapacity();  // capacity added by previous strategies _this round_
+    int currentDemand = snapshot.getQueueLength();
+    LOGGER.log(
+        Level.FINE,
+        "Available capacity={0}, currentDemand={1}",
+        new Object[] {availableCapacity, currentDemand});
+    if (availableCapacity < currentDemand) {
+      List<Cloud> jenkinsClouds = new ArrayList<>(Jenkins.get().clouds);
+      Collections.shuffle(jenkinsClouds);
+      for (Cloud cloud : jenkinsClouds) {
+        int workloadToProvision = currentDemand - availableCapacity;
+        if (!(cloud instanceof ComputeEngineCloud)) continue;
+        if (!cloud.canProvision(label)) continue;
+        ComputeEngineCloud gcp = (ComputeEngineCloud) cloud;
+        if (!gcp.isNoDelayProvisioning()) continue;
+        for (CloudProvisioningListener cl : CloudProvisioningListener.all()) {
+          if (cl.canProvision(cloud, strategyState.getLabel(), workloadToProvision) != null) {
+            continue;
+          }
+        }
+        Collection<NodeProvisioner.PlannedNode> plannedNodes =
+            cloud.provision(label, workloadToProvision);
+        LOGGER.log(Level.FINE, "Planned {0} new nodes", plannedNodes.size());
+        fireOnStarted(cloud, strategyState.getLabel(), plannedNodes);
+        strategyState.recordPendingLaunches(plannedNodes);
+        availableCapacity += plannedNodes.size();
+        LOGGER.log(
+            Level.FINE,
+            "After provisioning, available capacity={0}, currentDemand={1}",
+            new Object[] {availableCapacity, currentDemand});
+        break;
+      }
+    }
+    if (availableCapacity >= currentDemand) {
+      LOGGER.log(Level.FINE, "Provisioning completed");
+      return NodeProvisioner.StrategyDecision.PROVISIONING_COMPLETED;
+    } else {
+      LOGGER.log(Level.FINE, "Provisioning not complete, consulting remaining strategies");
+      return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+    }
+  }
+
+  private static void fireOnStarted(
+      final Cloud cloud,
+      final Label label,
+      final Collection<NodeProvisioner.PlannedNode> plannedNodes) {
+    for (CloudProvisioningListener cl : CloudProvisioningListener.all()) {
+      try {
+        cl.onStarted(cloud, label, plannedNodes);
+      } catch (Error e) {
```

Comment:
Could this be further specified or is this as much as would be known about the expected exceptions?

Reply:
This particular method is a copy from the Jenkins NodeProvisioner class:
- https://github.com/jenkinsci/jenkins/blob/450d87491df16598409d26511bcad02978eaa8bb/core/src/main/java/hudson/slaves/NodeProvisioner.java#L903-L916

That particular error design was done in https://github.com/jenkinsci/jenkins/pull/2591